examples/migraphx/custom_op_miopen_kernel/custom_op_miopen_kernel.cpp

```diff
@@ -0,0 +1,173 @@
+{
+    miopenActivationDescriptor_t ad;
+    MIGRAPHX_MIOPEN_ASSERT(miopenCreateActivationDescriptor(&ad));
+    miopenSetActivationDescriptor(ad, mode, alpha, beta, gamma);
+    return ad;
+}
+
+struct abs_custom_op final : migraphx::experimental_custom_op_base
+{
+    virtual std::string name() const override { return "abs_custom_op"; }
+    virtual migraphx::argument compute(migraphx::context ctx,
+                                       migraphx::shape output_shape,
+                                       migraphx::arguments args) const override
+    {
+        float alpha = 1;
+        float beta  = 0;
+        // MIOpen kernel call takes raw buffer pointers for the TensorData. These Buffer pointers
+        // must be accompanied with Tensor Description e.g. shape, type, strides, dimensionality.
+        // Following `make_miopen_tensor` makes such tensor descriptors to pass as parameter to
+        // MIOpen kernel call.
+        auto y_desc = make_miopen_tensor(output_shape);
+        auto x_desc = make_miopen_tensor(args[0].get_shape());
+        // create MIOpen stream handle
+        auto miopen_handle = make_miopen_handle(ctx);
+        // MIOpen has generic kernel for many different kinds of activation functions.
+        // Each such generic call must be accompanied with description of what kind of activation
+        // computation to perform
+        auto ad = make_activation_descriptor(miopenActivationABS, 0, 0, 0);
+        miopenActivationForward(
+            miopen_handle, ad, &alpha, x_desc, args[0].data(), &beta, y_desc, args[1].data());
+        return args[1];
+    }
+
+    virtual migraphx::shape compute_shape(migraphx::shapes inputs) const override
+    {
+        if(inputs.size() != 2)
+        {
+            throw std::runtime_error("abs_custom_op must have two input arguments");
+        }
+        if(inputs[0] != inputs[1])
+        {
+            throw std::runtime_error("Input arguments to abs_custom_op must have same shape");
+        }
+        return inputs.back();
+    }
+};
+
+int main(int argc, const char* argv[])
+{
+    abs_custom_op abs_op;
+    migraphx::register_experimental_custom_op(abs_op);
+    migraphx::program p;
+    migraphx::shape s{migraphx_shape_float_type, {32, 256}};
+    migraphx::module m = p.get_main_module();
+    auto x             = m.add_parameter("x", s);
+    auto neg_ins       = m.add_instruction(migraphx::operation("neg"), {x});
+    // add allocation for the custom_kernel's output buffer
+    auto alloc         = m.add_allocation(s);
+    auto custom_kernel = m.add_instruction(migraphx::operation("abs_custom_op"), {neg_ins, alloc});
+    auto relu_ins      = m.add_instruction(migraphx::operation("relu"), {custom_kernel});
+    m.add_return({relu_ins});
+
+    migraphx::compile_options options;
+    // set offload copy to true for GPUs
+    options.set_offload_copy();
+    p.compile(migraphx::target("gpu"), options);
+    migraphx::program_parameters pp;
```

Comment:
rename this to p_param, or something more descriptive hard to follow with program being p already.

Reply:
Changed it to `prog_params`.  `pp` is used in many other tests, that's why i'd used `pp` here as well. `pp` name is mostly used in limited context whenever we need to run the program. So it should be fine otherwise.